app/models/ability.rb

```diff
@@ -0,0 +1,34 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, Post
+    return unless user.present?
+
+    can :manage, Post, user_id: user.id
+    can :create, Comment
+    can :manage, Comment, user_id: user.id
+    can :create, Like
+    return unless user.is? 'admin'
```

Comment:
- [ ] [OPTIONAL] The code does work fine here but I would use if , else condition here